MerkleTreeAirdrop/contracts/MerkleTreeAirdrop.sol

```diff
@@ -1,24 +1,32 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import {MerkleDistributor} from "./MerkleDistributor.sol";
 import {IBunzz} from "./interfaces/IBunzz.sol";
 
-contract MerkleTreeAirdrop is MerkleDistributor {
+contract MerkleTreeAirdrop is IBunzz, Ownable, MerkleDistributor {
   using SafeERC20 for IERC20;
 
   event Claimed(uint256 index, address account, uint256 amount);
 
-  address public immutable token;
+  address public token;
 
   constructor(address token_, bytes32 merkleRoot_) {
     token = token_;
     __MerkleDistributor_init(merkleRoot_);
   }
 
+  function connectToOtherContracts(
+    address[] calldata contracts
+  ) external override onlyOwner {
+    require(contracts.length > 0, "No contract to connect");
+    token = contracts[0];
+  }
+
```

Comment:
looks good
but in the old code, token address was immutable
the owner can change the token address at anytime
I suggest you remove token_ input param in constructor function.
and once you add the token address using `connectToOtherContracts` function, the address shouldn't be changed.

Reply:
@deep-quality-dev @solcmoon 
I think owner can change the token address after all token balances are claimed...
So owner can use several times for airdrop, but before change token, we need to check the balance of the old token... So if the balance of the old token is greater than 0, it should not be changed.

Reply:
@Top-Kraken 
You have a point, but in this case, you should also write a code to reset the mapping variables.
Unless it is a service that specializes in airdrops to other dapps, there will be no need to update the token address several times.
So we can simply make it setting the address only once.

@deep-quality-dev 
remove the token param in constructor function and add this require in connectToOtherContracts function
`require(token != address(0x0), "token address was already set")`